openpype/hosts/equalizer/tests/test_plugin.py

```diff
@@ -0,0 +1,135 @@
+"""
+3dequalizer plugin tests
+
+These test need to be run in 3dequalizer.
+
+"""
+import json
+import re
+import unittest
+from pprint import pprint
+
+from attrs import asdict, define, field
+from attrs.exceptions import NotAnAttrsClassError
+
+AVALON_CONTAINER_ID = "test.container"
+
+CONTEXT_REGEX = re.compile(
+        r"AYON_CONTEXT::(?P<context>.*?)::AYON_CONTEXT_END",
+        re.DOTALL)
+
+
+@define
+class Container(object):
+
+    name: str = field(default=None)
+    id: str = field(init=False, default=AVALON_CONTAINER_ID)
+    namespace: str = field(default="")
+    loader: str = field(default=None)
+    representation: str = field(default=None)
+
+
+class Tde4Mock:
+    """Simple class to mock few 3dequalizer functions.
+
+    Just to run the test outside the host itself.
+    """
+
+    _notes = ""
+
+    def isProjectUpToDate(self):
+        return True
+
+    def setProjectNotes(self, notes):
+        self._notes = notes
+
+    def getProjectNotes(self):
+        return self._notes
+
+
+tde4 = Tde4Mock()
+
+
+def get_context_data():
+    m = re.search(CONTEXT_REGEX, tde4.getProjectNotes())
+    return json.loads(m["context"]) if m else {}
+
+
+def update_context_data(data, changes):
+    m = re.search(CONTEXT_REGEX, tde4.getProjectNotes())
+    if not m:
+        tde4.setProjectNotes(f"AYON_CONTEXT::::AYON_CONTEXT_END")
+    update = json.dumps(data, indent=4)
+    tde4.setProjectNotes(
+        re.sub(
+            CONTEXT_REGEX,
+            f"AYON_CONTEXT::{update}::AYON_CONTEXT_END",
+            tde4.getProjectNotes()
+        )
+    )
+
+
+def get_containers():
+    return get_context_data().get("containers", [])
+
+
+def add_container(container: Container):
+    context_data = get_context_data()
+    containers = get_containers()
```

Comment:
This should really be using the output of `get_context_data` since `get_containers` basically queries the same.